File: hw/tipan/u3/hw21/AnimalFarmSystemPython/model/Pig.py
from model.FarmAnimal import FarmAnimal
from controller.IMeatAnimal import IMeatAnimal


class Pig(FarmAnimal, IMeatAnimal):

    def __init__(self, ideal_weight, id, breed, born_on,
                 weight, slaughter_house, product, cuts):

        super().__init__(id, breed, born_on, weight,
                         slaughter_house, product, cuts)

        self.ideal_weight = ideal_weight

    def feed(self, food):
        print(f"Feeding THE PIG with food --> {food} in a bowl and water")

    def cut(self):
        return []

    def send_to_slaughter_house(self, slaughter_house):
        self.slaughter_house = slaughter_house

    def is_ready_for_slaughter(self):
        return self.weight >= self.ideal_weight

File: hw/tipan/u3/hw20FarmAnimalSystem/AnimalFarmSystem/src/main/java/ec/edu/espe/animalfarrmsystem/model/Pig.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package ec.edu.espe.animalfarrmsystem.model;

import ec.edu.espe.animalfarrmsystem.controller.IMeatAnimal;
import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author Ronald Tipan, The_Softwarriors ,@ESPE
 */
public class Pig extends FarmAnimal implements IMeatAnimal {
    private float idealWeight;

    @Override
    public void feed(Food food) {
        System.out.println("Feeding THE PIG with food -->" + food +"in a bowl and water");
    }
    @Override
    public ArrayList<Cut> cut() {
        ArrayList<Cut>cuts;
        cuts = new ArrayList<>();
        return cuts;
    }

    @Override
    public void sendToSlaughterHouse(SlaughterHouse slaughterHouse) {
        this.slaughterHouse = slaughterHouse; 
    }
    
    public Pig(float idealWeight, int id, String breed, Date bornOn, float weight, SlaughterHouse slaughterHouse, Product product, ArrayList<Cut> cuts) {
        super(id, breed, bornOn, weight, slaughterHouse, product, cuts);
        this.idealWeight = idealWeight;
    }
    
    /**
     * @return the idealWeight
     */
    public float getIdealWeight() {
        return idealWeight;
    }

    /**
     * @param idealWeight the idealWeight to set
     */
    public void setIdealWeight(float idealWeight) {
        this.idealWeight = idealWeight;
    }

    public boolean isReadyForSlaughter() {
    return getWeight() >= idealWeight;
    }
}
    